pkg/sync/kubernetes/featureflagconfiguration/clientset.go

```diff
@@ -91,7 +92,7 @@ func WatchResources(ctx context.Context, l log.Entry, clientSet FFCInterface, re
 			},
 		},
 		&ffv1alpha1.FeatureFlagConfiguration{},
-		refreshTime,
+		0,
```

Comment:
No need to poll every x seconds

Reply:
That's outside the scope of this change really, the polling should be configurable because it puts quite a lot of load on the API server

Reply:
> - resyncPeriod: if non-zero, will re-list this often (you will get OnUpdate
calls, even if nothing changed). Otherwise, re-list will be delayed as
long as possible (until the upstream source closes the watch or times out,
or you stop the controller).

I've gone into a rabbit hole of trying to understand what the re-sync does. Digging around it seems like a re-sync re-fires events from the cached queue (hence why we're warned of OnUpdate calls, even if nothing changed). Based on this assumption I think that having no re-sync period reduces load as it avoids the re-list and re-firing of events.
https://groups.google.com/g/kubernetes-sig-api-machinery/c/PbSCXdLDno0?pli=1
https://github.com/kubernetes-client/java/issues/725#issuecomment-540352746

What do you think?

Reply:
Happy to take your word for it, we actually need to do some proper performance testing and until then we should build it to our best ability then we can take stock of the tuning.

Reply:
I think for now we make it configurable via the `ProviderArgs` system, what do you think?

Reply:
> Just looking at that thread give me some pauses for concern
> 
> > Recommendations found in the wild as to the right resync period seem to gravitate towards "keep it rather high to avoid burdening the API server too much", and I often see values on the order of multiple minutes or even hours. However, this advice seems to render the resync period unsuitable as a means to compensate for watch errors for somewhat latency-critical controllers.

From my understanding that advice is if you're setting the refresh sync period to a non zero value.
I've just stepped into the code to confirm that having a non-zero re-sync period performs more work than zero:
https://github.com/kubernetes/client-go/blob/master/tools/cache/reflector.go#L267

I think leaving it configurable is good too but I'd opt for the default as 0, what do you reckon?

Reply:
> If you've confirmed this via debugger, it seems like the right choice to me. It's too bad that sort of detective work was necessary.

Poor choice of words from me, I didn't use a debugger but it's relatively easy to see the code path in this case thankfully (I wish I'd gone through the code earlier).

Reply:
I agree, make it configurable as a parameter but set the default to zero

Reply:
Done